Visualize Session Types and Terms

jrjohnson opened the pull request:
Adds click throughs to session types and to terms that lead to different ways of seeing how terms and sessions types intersect in a course.

Fixes #3126
Fixes #3127

jrjohnson commented:
Deploy preview for *ilios-frontend* ready!

Built with commit 609fc8cb46c9d97b11e22a27e0f4f4b0ab359d65

https://deploy-preview-3644--ilios-frontend.netlify.com

saschaben commented:
Initial feedback: 

- activating the clickthru is a little wonky, seems poorly mapped to the chart items...
- Better title labeling. for Session Type Chart: `Vocabulary Terms for [Session Type]`. for vocabulary visualization: `Session Types for [Vocabulary Name] - [Term]`.
- Better vocabluary term labeling for the clickthru in session type. Should include the parent Vocabulary name in the bar (eg, `Topics - Anatomy  16.2%` rather than `Anatomy  16.2%`)

saschaben commented:
also: with a 2nd layer of charts, the "Other Visualizations" back button now is a bit misleading, since it is really just a history(-1) not a link back to the visualizations landing page.

saschaben commented:
...and is it possible to add a timeout for the tooltips, so that they vanish after a set time, and reqiure a reactivate of mouseover to show up again? I'm thinking along the lines of the metatext for xkcd if you want an example...

jrjohnson commented:
@saschaben this is ready for review again. I fixed all outstanding issues and rebased it with changes from ember-simple-charts so clicking and tooltips should be much improved.

saschaben commented:
I'm still getting a few edge case oddities for instance, at https://deploy-preview-3644--ilios-frontend.netlify.com/data/courses/1194/session-types/19 we get gaps between categories, but we do not get them with any other value other than "independent learning"...also on the same page, if my browser is not full screen, I cannot seem to get the tooltip for the first item in each category to pop up. With full screen activated, this issue vanishes.

Otherwise, this looks great. Still lots of things we can do in future iterations, but I think this PR is pretty much there.
